test/metabase/driver/util_test.clj

```diff
@@ -239,15 +239,17 @@
     (is (true? (driver.u/semantic-version-gte [5 0] [4 0])))
     (is (true? (driver.u/semantic-version-gte [5 0 1] [4 0])))
     (is (true? (driver.u/semantic-version-gte [5 0] [4 0 1])))
-    (is (true? (driver.u/semantic-version-gte [4 0] [4 0 1])))
     (is (true? (driver.u/semantic-version-gte [5 0] [4 1])))
     (is (true? (driver.u/semantic-version-gte [4 1] [4 1])))
     (is (true? (driver.u/semantic-version-gte [4 1] [4])))
     (is (true? (driver.u/semantic-version-gte [4] [4])))
+    (is (true? (driver.u/semantic-version-gte [4] [4 0 0])))
     (is (true? (driver.u/semantic-version-gte [4 nil] [4])))
     (is (false? (driver.u/semantic-version-gte [3] [4])))
     (is (false? (driver.u/semantic-version-gte [3] [4 nil])))
     (is (false? (driver.u/semantic-version-gte [4] [4 1])))
     (is (false? (driver.u/semantic-version-gte [4 nil] [4 1])))
+    (is (false? (driver.u/semantic-version-gte [4 0] [4 0 1])))
     (is (false? (driver.u/semantic-version-gte [4 0 1] [4 1])))
-    (is (false? (driver.u/semantic-version-gte [3 9] [4 0])))))
+    (is (false? (driver.u/semantic-version-gte [3 9] [4 0])))
+    (is (false? (driver.u/semantic-version-gte [3 1] [4])))))
```

Comment:
The original version returned `true` in this case.

Reply:
Perhaps we should tests for `nil` arguments? Maybe this should be an assertion and throw an error?